zookeeper-server/src/main/java/org/apache/zookeeper/ClientCnxn.java

```diff
@@ -928,12 +925,12 @@ else if (serverPath.length() > chrootPath.length())
                     packet.replyHeader.setErr(
                             KeeperException.Code.CONNECTIONLOSS.intValue());
                     throw new IOException("Xid out of order. Got Xid "
-                            + replyHdr.getXid() + " with err " +
-                            + replyHdr.getErr() +
-                            " expected Xid "
+                            + replyHdr.getXid() + " with err "
+                            + +replyHdr.getErr()
```

Comment:
This `+ +` can be cleaned up.

Reply:
Here is a rule `+' should be on a new line`. For cleanup I guess you mean replace it with `String.format`? It would be a follow up to unify the log/Exception format and generation like that for javadoc(ZOOKEEPER-3469). This pull request would be a clean start point for all of this request but we should prevent it from too complex IMHO.

Reply:
In this specific case, there are two uses of the `+` operator on the line. It's exactly what the old code has so your change preserves the original mistake - it's just the kind of thing that I'm surprised compiles.